reconcile/external_resources/manager.py

```diff
@@ -0,0 +1,423 @@
+    ) -> ReconcileStatus:
+        pass
+
+    @abstractmethod
+    def reconcile_resource(
+        self,
+        image: str,
+        key: ExternalResourceKey,
+        input: str,
+        action: Action,
+        dry_run: bool,
+    ) -> None:
+        pass
+
+
+class ExternalResourcesJobImpl(ExternalResourcesImpl):
+    def __init__(
+        self, ri: ResourceInventory, oc: OCCli, cluster: str, namespace: str
+    ) -> None:
+        self.ri = ri
+        self.cluster = cluster
+        self.namespace = namespace
+        self.oc = oc
+
+    def _get_job_name(self, key: ExternalResourceKey) -> str:
+        # TODO: This needs to be changed to include other fields or a hash
+        return key.identifier
+
+    def _build_job_spec(
+        self, key: ExternalResourceKey, image: str, input: str, action: Action
+    ) -> dict[str, Any]:
+        job = {
+            "apiVersion": "batch/v1",
+            "kind": "Job",
+            "metadata": {
+                "name": self._get_job_name(key),
+                "annotations": {
+                    "provision_provider": key.provision_provider,
+                    "provisioner": key.provisioner_name,
+                    "provider": key.provision_provider,
+                    "identifier": key.identifier,
+                },
+            },
+            "spec": {
+                "template": {
+                    "spec": {
+                        "initContainers": [
+                            {
+                                "name": "job",
+                                "image": image,
+                                "imagePullPolicy": "Always",
+                                "env": [
+                                    {
+                                        "name": "DRY_RUN",
+                                        "value": "False",
+                                    },
+                                    {
+                                        "name": "INPUT",
+                                        "value": input,
+                                    },
+                                    {
+                                        "name": "ACTION",
+                                        "value": action,
+                                    },
+                                ],
+                                "volumeMounts": [
+                                    {
+                                        "name": "credentials",
+                                        "mountPath": "/credentials",
+                                        "subPath": "credentials",
+                                    },
+                                    {
+                                        "name": "workdir",
+                                        "mountPath": "/work",
+                                    },
+                                ],
+                            }
+                        ],
+                        "containers": [
```

Comment:
Nice idea to use an init-container for the actual work and a std container for outputs.